codegen/validation.go

```diff
@@ -197,34 +195,22 @@ func ValidationCode(att *expr.AttributeExpr, req, ptr, def bool, target, context
 
 // RecursiveValidationCode produces Go code that runs the validations defined in
 // the given attribute and its children recursively against the value held by
-// the variable named target. See ValidationCode for a description of the
-// arguments and their effects.
-func RecursiveValidationCode(att *expr.AttributeExpr, req, ptr, def bool, target string) string {
-	seen := make(map[string]*bytes.Buffer)
-	return recurseValidationCode(att, req, ptr, def, target, target, seen).String()
-}
-
-func hasValidationsRecurse(att *expr.AttributeExpr, ignoreRequired bool, seen map[*expr.AttributeExpr]struct{}) bool {
-	if att.Validation != nil {
-		if !ignoreRequired || !att.Validation.HasRequiredOnly() {
-			return true
-		}
-	}
-	if o := expr.AsObject(att.Type); o != nil {
-		for _, catt := range *o {
-			if _, ok := seen[catt.Attribute]; ok {
-				continue // break infinite recursions
-			}
-			seen[catt.Attribute] = struct{}{}
-			if hasValidationsRecurse(catt.Attribute, ignoreRequired, seen) {
-				return true
-			}
-		}
+// the variable named target.
+//
+// ahs is the optional AttributeHelper to use to compute validation. If none
```

Comment:
Would it be cleaner to make `ahs` just a normal - non variadic argument and have callers instantiate the default attribute analyzer if non other is given?

Reply:
Fixed